### PR TITLE
[FEATURE] cpt-city catalog support in color ramp's gradient-only mode

### PR DESCRIPTION
**Pull request by nirvn**
This PR brings cpt-city catalog ramps to:
- shapeburst fill
- gradient fill
- inner/outer glow effect

Ugly proof of life:
fixed

### CODE REVIEW

File: src/core/effects/qgsgloweffect.cpp [hunk 1]
```diff
@@ -176,7 +176,14 @@ void QgsGlowEffect::readProperties( const QgsStringMap &props )
 
   //attempt to create color ramp from props
   delete mRamp;
-  mRamp = QgsGradientColorRamp::create( props );
+  if ( props.contains( QStringLiteral( "schemeName" ) ) )
```
Comment (nyalldawson) [inline src/core/effects/qgsgloweffect.cpp]: This check looks a bit fragile - can we instead have a proper property for the ramp type?

Comment (nirvn) [inline src/core/effects/qgsgloweffect.cpp]: @nyalldawson , for ramps to add a "type" property in the qstringmap?

Comment (nyalldawson) [inline src/core/effects/qgsgloweffect.cpp]: Yeah. Otherwise it'll get very messy if additional gradient type ramps are added.

**Comment by nirvn** [comment]
@nyalldawson , PR updated to rely on a (newly-added) rampType property.

**Comment by nyalldawson** [comment]
+1

**Pull request merged**